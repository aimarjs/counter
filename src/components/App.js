import React, { Component } from 'react';
import styled from 'styled-components';

import Button from './Button'

const initialState = {
  counter: 0
}

class App extends Component {
  state = initialState

  addHandler = () => {
    const currentState = this.state.counter;

    this.setState({ counter: currentState + 1})
  }

  removeHandler = () => {
    const currentState = this.state.counter;

    if (currentState !== 0) {
      this.setState({ counter: currentState - 1 })
    }
  }

  resetHandler = () => {

    this.setState(initialState)
  }

  render() {
    return (
      <AppWrapper>
        <HeaderWrapper>
          <Counter>{this.state.counter}</Counter>
        </HeaderWrapper>
        <BodyWrapper>
          <Controls>
            <Button title="Add" clickHandler={this.addHandler} />
            <Button title="Reset" clickHandler={this.resetHandler} />
            <Button title="Remove" clickHandler={this.removeHandler} />
          </Controls>
        </BodyWrapper>
      </AppWrapper>
    );
  }
}

const AppWrapper = styled.div`
  width: 100vw;
  height: 100vh;
  margin: 0;
  padding: 0;
`

const HeaderWrapper = styled.div`
  text-align: center;
`

const BodyWrapper = styled.div`
  text-align: center;
`

const Controls = styled.div`
  margin: 0 auto;
  max-width: 300px;
`

const Counter = styled.h2`
  font-size: 64px;
`

export default App;
