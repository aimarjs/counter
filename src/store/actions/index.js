import shortId from 'shortid';
import * as actionTypes from './actionTypes';

export const addNewEntry = data => ({
  type: actionTypes.ADD_NEW_ENTRY,
  payload: data,
  id: shortId.generate(),
});

export const getAllEntries = () => ({
  type: actionTypes.GET_ALL_ENTRIES,
});

export const addHandler = () => ({
  type: actionTypes.INC_COUNTER,
});

export const removeHandler = () => ({
  type: actionTypes.DEC_COUNTER,
});

export const resetHandler = () => ({
  type: actionTypes.RESET_COUNTER,
});
