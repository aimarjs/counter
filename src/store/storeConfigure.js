import thunk from 'redux-thunk';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import counterReducer from './reducers/counterReducer'

const rootReducer = combineReducers({
  counter: counterReducer
})

const composeEnhancers = composeWithDevTools(applyMiddleware(thunk))

const store = createStore(
  rootReducer,
  composeEnhancers
)

// const store = createStore(rootReducer, composeEnhancers);

export default store;
