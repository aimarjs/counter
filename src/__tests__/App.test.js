import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';

import App from '../components/App';
import store from '../store/storeConfigure';

describe('<App>', () => {

  it('renders the page', () => {
    const wrapper = shallow(
      <Provider store={store}><App /></Provider>
    )
    expect(wrapper).toMatchSnapshot();
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


})
