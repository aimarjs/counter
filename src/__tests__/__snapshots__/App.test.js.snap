// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App capturing snapshot of App 1`] = `<Connect(App) />`;
